modules/ROOT/generate-docs.sh

```diff
@@ -21,14 +49,25 @@ generate() {
         --dereferenced \
         --output "${PATH_TO_BUNDLE}/bundled-${WHAT}.yaml"
 
+    # Get the bundled spec, apply the OpenAPI overlay,
+    # and output to the temporary bld folder.
+    npx openapi-overlays-js \
+        --openapi "${PATH_TO_BUNDLE}/bundled-${WHAT}.yaml" \
+        --overlay "${PATH_TO_OVERLAY}/${WHAT}.yaml" \
+        > "${PATH_TO_TEMP}/${WHAT}.yaml"
+
+    # Build the reference documentation using the overlaid spec.
     npx @redocly/cli \
       build-docs \
       --config redocly.yaml \
-      "${PATH_TO_SYNC_GATEWAY}/docs/api/${WHAT}.yaml" \
+      "${PATH_TO_TEMP}/${WHAT}.yaml" \
       --output partials/sgw-openapi-$WHAT.html \
       --template partials/redocAll.hbs
 }
 
 generate public
```

Comment:
I wonder if your local .redocly.yaml needs to extend the .redocly.yaml that is in sync_gateway repo? then it pulls in the code that we have to strip internal APIs.

For capella, this is even more important because it will run some javascript code to process out capella-only information.

Reply:
Well, we're already using the `.redocly.yaml` that is in the `sync_gateway` repo when we create the bundled spec, as the first step in the `generate()` routine. However you are right, it is important that we need to run all the filters that the upstream `.redocly.yaml` specifies when we create the bundled spec. So I've updated the bundle step to specify the API by the name assigned in the `apis` section of the upstream `.redocly.yaml`.